src/WebJobs.Script/Rpc/LanguageWorkerChannel.cs

```diff
@@ -120,18 +133,143 @@ internal void StartWorker()
+                if (_process.ExitCode != 0)
+                {
+                    var processExitEx = new LanguageWorkerProcessExitException($"{_process.StartInfo.FileName} exited with code {_process.ExitCode}\n {exceptionMessage}");
+                    HandleWorkerError(processExitEx);
+                    workerErrorHandled = true;
+                }
+                else
+                {
+                    _process.WaitForExit();
+                    _process.Close();
+                }
+            }
+            catch (Exception ex)
+            {
+                if (!workerErrorHandled)
+                {
+                    var processExitEx = new LanguageWorkerProcessExitException($"Worker process is not attached. {exceptionMessage}", ex);
+                    HandleWorkerError(processExitEx);
+                }
+            }
+        }
+
+        private void OnErrorDataReceived(object sender, DataReceivedEventArgs e)
+        {
+            // TODO: per language stdout/err parser?
+            if (e.Data != null)
+            {
+                string msg = e.Data;
+                if (msg.IndexOf("warn", StringComparison.OrdinalIgnoreCase) > -1)
+                {
+                    if (IsLanguageWorkerConsoleLog(msg))
+                    {
+                        msg = RemoveLogPrefix(msg);
+                        _workerChannelLogger?.LogWarning(msg);
+                    }
+                    _userLogsConsoleLogger?.LogWarning(msg);
+                }
+                else if ((msg.IndexOf("error", StringComparison.OrdinalIgnoreCase) > -1) ||
+                          (msg.IndexOf("fail", StringComparison.OrdinalIgnoreCase) > -1) ||
+                          (msg.IndexOf("severe", StringComparison.OrdinalIgnoreCase) > -1))
+                {
+                    if (IsLanguageWorkerConsoleLog(msg))
+                    {
+                        msg = RemoveLogPrefix(msg);
+                        _workerChannelLogger?.LogError(msg);
+                    }
+                    _userLogsConsoleLogger?.LogError(msg);
+                    AddStdErrMessage(Sanitizer.Sanitize(msg));
+                }
+                else
+                {
+                    if (IsLanguageWorkerConsoleLog(msg))
+                    {
+                        msg = RemoveLogPrefix(msg);
+                        _workerChannelLogger?.LogInformation(msg);
+                    }
+                    _userLogsConsoleLogger?.LogInformation(msg);
+                }
+            }
+        }
+
+        internal void AddStdErrMessage(string msg)
+        {
+            if (_processStdErrDataQueue.Count >= _maxNumberOfErrorMessages)
+            {
+                _processStdErrDataQueue.Dequeue();
+                _processStdErrDataQueue.Enqueue(msg);
+            }
+            else
+            {
+                _processStdErrDataQueue.Enqueue(msg);
+            }
+        }
+
+        internal bool IsLanguageWorkerConsoleLog(string msg)
```

Comment:
Tracking improvement to system logs coming from language workers here: https://github.com/Azure/azure-functions-host/issues/3117